flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/batch/sql/RowLevelUpdateTest.java

```diff
@@ -0,0 +1,149 @@
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.planner.plan.batch.sql;
+
+import org.apache.flink.table.api.ExplainDetail;
+import org.apache.flink.table.api.TableConfig;
+import org.apache.flink.table.api.config.ExecutionConfigOptions;
+import org.apache.flink.table.connector.sink.abilities.SupportsRowLevelUpdate;
+import org.apache.flink.table.planner.utils.BatchTableTestUtil;
+import org.apache.flink.table.planner.utils.JavaScalaConversionUtil;
+import org.apache.flink.table.planner.utils.TableTestBase;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+
+import scala.collection.Seq;
+
+/** Test for row-level update. */
+@RunWith(Parameterized.class)
+public class RowLevelUpdateTest extends TableTestBase {
+
+    private final Seq<ExplainDetail> explainDetails =
+            JavaScalaConversionUtil.toScala(
+                    Collections.singletonList(ExplainDetail.JSON_EXECUTION_PLAN));
+    private final SupportsRowLevelUpdate.RowLevelUpdateMode updateMode;
+
+    private BatchTableTestUtil util;
+
+    @Parameterized.Parameters(name = "updateMode = {0}")
+    public static Collection<SupportsRowLevelUpdate.RowLevelUpdateMode> data() {
+        return Arrays.asList(
+                SupportsRowLevelUpdate.RowLevelUpdateMode.UPDATED_ROWS,
+                SupportsRowLevelUpdate.RowLevelUpdateMode.ALL_ROWS);
+    }
+
+    public RowLevelUpdateTest(SupportsRowLevelUpdate.RowLevelUpdateMode updateMode) {
+        this.updateMode = updateMode;
+    }
+
+    @Before
+    public void before() {
+        util = batchTestUtil(TableConfig.getDefault());
+        util.tableEnv()
+                .getConfig()
+                .getConfiguration()
+                .set(ExecutionConfigOptions.TABLE_EXEC_RESOURCE_DEFAULT_PARALLELISM, 12);
+    }
+
+    @Test
+    public void testUpdateWithoutFilter() {
+        createTableForUpdate();
+        util.verifyExplainInsert("UPDATE t SET b = 'n1', a = char_length(b) * a ", explainDetails);
+    }
+
+    @Test
+    public void testUpdateWithFilter() {
+        createTableForUpdate();
+        util.verifyExplainInsert(
+                "UPDATE t SET b = 'v2' WHERE a = 123 AND b = 'v1'", explainDetails);
+    }
+
+    @Test
+    public void testUpdateWithSubQuery() {
+        createTableForUpdate();
+        util.tableEnv()
+                .executeSql(
+                        String.format(
+                                "CREATE TABLE t1 (a int, b string) WITH "
+                                        + "('connector' = 'test-update-delete', 'update-mode' = '%s') ",
+                                updateMode));
+        util.verifyExplainInsert(
+                "UPDATE t SET b = 'v2' WHERE a = (SELECT count(*) FROM t1)", explainDetails);
+    }
+
+    @Test
+    public void testUpdateWithOnlyRequireUpdatedCols() {
```

Comment:
add a new case which not contains all columns in update clause, e.g., a source table with more columns here 'TABLE t (f0, f1,..., a int, b string, c double, ...)'